### PR TITLE
Menu actions

### PR DESCRIPTION
**Pull request by Carreau**
Draft/Proof of concept. Done in the car while moving from one flat to another (no I was not driving).

So what are you suppose to see here. 
Purely declarative menu form: 

```
"Edit",[
                'ipython.cut-selected-cell',
                'ipython.copy-selected-cell',
                'ipython.paste-cell-before',
                'ipython.paste-cell-after',
                '<divider>',
                'ipython.delete-cell',
                '<divider>',
                ['Submenu',[
                    'Action 1',
                    'Action 2',
                    'Action 3'
                    ]
                ]
```

And automatically show icon/shortcut if associated with action. 

capture d ecran 2014-12-22 a 18 29 49

Still highly work in progress.

### CODE REVIEW

**Comment by minrk** [comment]
@Carreau planning to finish this one? I started to work on something, and realized it would be better for this to be merged first.

File: IPython/html/static/notebook/js/menubar.js [hunk 1]
```diff
@@ -39,6 +39,101 @@ define([
...
         this.save_widget = options.save_widget;
         this.quick_help = options.quick_help;
+        this.actions = options.actions;
+        if(!options.noinit){
+            this.init()
+        }
+        Object.seal(this);
+        
+
+        this.addMenu("Edit",[
```
Comment (minrk) [inline IPython/html/static/notebook/js/menubar.js]: so we're using camelCase now?

**Comment by Carreau** [comment]
Still moving, and heading to us embassy tomorrow morning so limited bandwidth. So feel free to move forward. 

Envoyé de mon iPhone

**Comment by Carreau** [comment]
Marking pessimistically as 4.0.

**Comment by minrk** [comment]
I'll pick this one up, you focus on visa stuff.

File: IPython/html/static/notebook/js/menubar.js (hunk 1 above)
Comment (jdfreder) [inline IPython/html/static/notebook/js/menubar.js]: :-1:

**Comment by minrk** [comment]
@Carreau how were you planning to handle the fact that most things in the menu are not actions? Were you planning to add a bunch of actions, or have a bunch of pseudo_actions here, like in the toolbar?

**Comment by Carreau** [comment]
I was planning to move some things to actions when it make sens (toggle all outputs... ) and add pseudo action for the rest (for now) something like `<#aname>` could be special cased to only append the div, with the `aname` id, and let the rest of the code works for now.

**Comment by takluyver** [comment]
From the dev meeting: this is a nice thing, but not critical, and it's tricky to check exhaustively for bugs that it may introduce, so we're punting it to 4.x in the interests of actually making a 3.0 release.

**Comment by Carreau** [comment]
I'll close and open an issue to redo it.

File: IPython/html/static/notebook/js/menubar.js (hunk 1 above)
Comment (jdfreder) [inline IPython/html/static/notebook/js/menubar.js]: heh, my opinion has since changed :+1:

**Pull request closed without merging**